server/src/test/java/server/haengdong/support/extension/DatabaseCleaner.java

```diff
@@ -0,0 +1,36 @@
+package server.haengdong.support.extension;
+
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.PersistenceContext;
+import java.util.List;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+@Component
+class DatabaseCleaner {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Transactional
+    public void clear() {
+        em.clear();
+        truncate();
+    }
+
+    private void truncate() {
+        em.createNativeQuery("SET REFERENTIAL_INTEGRITY FALSE").executeUpdate();
+        getTruncateQueries().forEach(query -> em.createNativeQuery(query).executeUpdate());
+        em.createNativeQuery("SET REFERENTIAL_INTEGRITY TRUE").executeUpdate();
+    }
+
+    private List<String> getTruncateQueries() {
+        String sql = """
+                SELECT Concat('TRUNCATE TABLE ', TABLE_NAME, ' RESTART IDENTITY', ';') AS q
+                FROM INFORMATION_SCHEMA.TABLES
+                WHERE TABLE_SCHEMA = 'PUBLIC'
+                """;
+
+        return em.createNativeQuery(sql).getResultList();
+    }
```

Comment:
```suggestion
     private List<String> tableNames;
     
     @PostConstruct
     public void getTableNames() {
                String sql = """
                SELECT TABLE_NAME
                FROM INFORMATION_SCHEMA.TABLES
                WHERE TABLE_SCHEMA = 'PUBLIC'
                """;

        return em.createNativeQuery(sql).getResultList();
    }
        
    @Transactional
    public void clear() {
            em.clear();
            truncate();
    }

    private void truncate() {
        em.createNativeQuery("SET REFERENTIAL_INTEGRITY FALSE").executeUpdate();
        tableNames.forEach(tableName -> em.createNativeQuery("TRUNCATE TABLE " + tableName +  "RESTART IDENTITY;").executeUpdate());
        em.createNativeQuery("SET REFERENTIAL_INTEGRITY TRUE").executeUpdate();
    }
```
개행이 조금 이상한데 이런 방식으로 빈 등록 후 한번만 날려도 될 것 같습니다.

Reply:
r: 쿼리문을 상수로 선언해놓고 재사용 하는 것이 좋을 것 같습니다.
그리고 MySQL 기준 테이블 truncate 하면 auto_increment가 초기화되기 때문에 restart identity 지워도 되겠네요.

```java
static final String referentialFormat = "SET REFERENTIAL_INTEGRITY %b;";
static final String truncateFormat = "TRUNCATE TABLE %s;"

// ... 

private void truncate() {
    em.createNativeQuery(String.format(referentialFormat, false)).executeUpdate();
    tableNames.forEach(tableName -> em.createNativeQuery(String.format(truncateForamt, tableName)).executeUpdate());
    em.createNativeQuery(String.format(referentialFormat, true)).executeUpdate();
}
```